doc/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,374 @@
+```
+
+There are two ways to run the operator:
+- As Go program outside a cluster
+- As a Deployment inside a Kubernetes cluster
+
+### 1. Run locally outside the cluster
+
+To run the operator locally execute the following command:
+
+```
+make run ENABLE_WEBHOOKS=false
+```
+
+```sh
+$ make run ENABLE_WEBHOOKS=false
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+go fmt ./...
+go vet ./...
+```
+
+### 2. Run as a Deployment inside the cluster
+
+Build the operator image and push it to a registry.
+
+```sh
+$ make docker-build docker-push IMG=<some-registry>/<project-name>:tag
+```
+
+To deploy the operator using an image, run:
+
+```sh
+$ make deploy IMG=<some-registry>/<project-name>:tag
+```
+
+Verify that the memcached-operator is up and running:
+```sh
+$ kubectl get deployment --all-namespaces
+NAMESPACE     NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
+default       controller-manager   1/1     	1            	1       48m
+
+$ kubectl get pods --all-namespaces
+NAMESPACE     NAME                             READY   STATUS    RESTARTS   AGE
+default    controller-manager-7fbdc4484c-ch999   1/1   Running   0          50m
+```
+
+### Create a Custom Resource
+
+Create a custom resource similar to [`cache.example.com_v1alpha1_memcached_cr.yml`][memcached_cr] inside `config/samples/` of kubebuilder project.
+
+```sh
+$ cat config/samples/cache.example.com_v1alpha1_memcached_cr.yml
+apiVersion: cache.example.com/v1alpha1
+kind: Memcached
+metadata:
+  name: memcached-sample
+spec:
+  # Add fields here
+  size: 3
+
+$ kubectl apply -f config/samples/cache.example.com_v1alpha1_memcached_cr.yml
+```
+
+Ensure that the memcached-operator creates the deployment for the CR:
+
+```sh
+$ kubectl get deployment --all-namespaces
+NAMESPACE     NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
+default       controller-manager   1/1        1             1          50m
+default       memcached-sample     3/3        3             3          51m
+```
+
+Check the pods and CR status to confirm the status is updated with the memcached pod names:
+
+```sh
+kubectl get pods --all-namespaces
+NAMESPACE     NAME                               READY   STATUS  RESTARTS   AGE
+default       controller-manager-7fbdc4484c-ch999 1/1  Running   0          50m
+default       memcached-sample-68b656fbd4-kbrcx   1/1  Running   0          51m
+default       memcached-sample-68b656fbd4-n9pf5   1/1  Running   0          51m
+default       memcached-sample-68b656fbd4-nlcpw   1/1  Running   0          51m
+```
+
+### Cleanup
+
+To delete CRDs from the cluster:
+
+```sh
+$ make uninstall 
+```
+
+Clean up the resources
+```sh
+$ kubectl delete -f config/samples/cache.example.com_v1alpha1_memcached_cr.yml
+```
```

Comment:
Removing the CRDs in the first step causes Kubernetes garbage collection to delete the CRs of that type first, if I'm not mistaken.

So I think these steps should be reversed because, by the time the second step runs the CR will have already been deleted (and that command might actually fail since the object does not exist).

Reply:
Yeah, that's a mistake. Ill change that.